test/test_macho.rb

```diff
@@ -46,6 +46,15 @@ def test_load_commands
     end
   end
 
+  def test_load_command_string_malformed
+    filename = fixture(:i386, "hello.bin")
+    file = MachO::MachOFile.new(filename)
+
+    assert_raises MachO::LCStrMalformedError do
+      file.change_install_name(file.linked_dylibs.first, "\x00bad")
```

Comment:
Yep, that's the behavior that I was counting on. Now that we're removing this, this test probably won't be of much user either...